components/raftstore/src/store/worker/refresh_config.rs

```diff
@@ -0,0 +1,214 @@
+        }
+    }
+}
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub enum ThreadPool {
+    Store,
+    Apply,
+}
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub enum Task {
+    ScalePool(ThreadPool, usize),
+}
+
+impl Display for Task {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match &self {
+            Task::ScalePool(pool, size) => {
+                write!(f, "Scale pool ")?;
+                match pool {
+                    ThreadPool::Store => write!(f, "ajusts apply: {} ", size),
+                    ThreadPool::Apply => write!(f, "ajusts raft: {} ", size),
+                }
+            }
+        }
+    }
+}
+
+pub struct Runner<EK, ER, AH, RH>
+where
+    EK: engine_traits::KvEngine,
+    ER: engine_traits::RaftEngine,
+    AH: HandlerBuilder<ApplyFsm<EK>, ControlFsm>,
+    RH: HandlerBuilder<PeerFsm<EK, ER>, StoreFsm<EK>>,
+{
+    apply_pool: PoolController<ApplyFsm<EK>, ControlFsm, AH>,
+    raft_pool: PoolController<PeerFsm<EK, ER>, StoreFsm<EK>, RH>,
+}
+
+impl<EK, ER, AH, RH> Runner<EK, ER, AH, RH>
+where
+    EK: engine_traits::KvEngine,
+    ER: engine_traits::RaftEngine,
+    AH: HandlerBuilder<ApplyFsm<EK>, ControlFsm>,
+    RH: HandlerBuilder<PeerFsm<EK, ER>, StoreFsm<EK>>,
+{
+    #[allow(dead_code)]
+    pub fn new(
+        apply_router: BatchRouter<ApplyFsm<EK>, ControlFsm>,
+        raft_router: BatchRouter<PeerFsm<EK, ER>, StoreFsm<EK>>,
+        apply_pool_state: PoolState<ApplyFsm<EK>, ControlFsm, AH>,
+        raft_pool_state: PoolState<PeerFsm<EK, ER>, StoreFsm<EK>, RH>,
+    ) -> Self {
+        let apply_pool = PoolController::new(apply_router, apply_pool_state);
+        let raft_pool = PoolController::new(raft_router, raft_pool_state);
+
+        Runner {
+            apply_pool,
+            raft_pool,
+        }
+    }
+
+    fn resize_raft_pool(&mut self, size: usize) {
+        let current_pool_size = self.raft_pool.state.expected_pool_size;
+        self.raft_pool.state.expected_pool_size = size;
+        match current_pool_size.cmp(&size) {
+            std::cmp::Ordering::Greater => self.raft_pool.decrease_by(current_pool_size - size),
+            std::cmp::Ordering::Less => self.raft_pool.increase_by(size - current_pool_size),
+            std::cmp::Ordering::Equal => return,
+        }
+        self.raft_pool.cleanup_poller_threads();
+        info!(
+            "resize raft pool";
+            "from" => current_pool_size,
+            "to" => self.raft_pool.state.expected_pool_size
+        );
+    }
+
+    fn resize_apply_pool(&mut self, size: usize) {
+        let current_pool_size = self.apply_pool.state.expected_pool_size;
+        self.apply_pool.state.expected_pool_size = size;
+        match current_pool_size.cmp(&size) {
+            std::cmp::Ordering::Greater => self.apply_pool.decrease_by(current_pool_size - size),
+            std::cmp::Ordering::Less => self.apply_pool.increase_by(size - current_pool_size),
+            std::cmp::Ordering::Equal => return,
```

Comment:
Perhaps adding some logs for this case could be helpful

Reply:
Changed to () operation, reuse the logs below